tiny_dnn/core/kernels/global_avepool_grad_op.h

```diff
@@ -27,10 +28,17 @@ class GlobalAvePoolGradOp : public core::OpKernel {
     // initialize outputs
     fill_tensor(prev_delta, float_t{0});
 
-    // only internal kernel op implemented yet, so use it regardless
-    // of the specified backend engine
-    kernels::global_avepool_grad_op_internal(prev_delta, curr_delta, params,
-                                             context.parallelize());
+    const core::backend_t engine = context.engine();
+
+    if (engine == core::backend_t::avx) {
+#ifdef CNN_USE_AVX
```

Comment:
@Randl I think killing `core::backend_t::avx` entry and all the AVX related codes when `CNN_USE_AVX` isn't defined would be a simpler solution.